packages/sprotty/src/features/context-menu/mouse-listener.ts

```diff
@@ -46,14 +47,28 @@ export class ContextMenuMouseListener extends MouseListener {
             return;
         }
         const mousePosition = { x: event.x, y: event.y };
+        const root = target.root;
+        const id = target.id;
         let isTargetSelected = false;
         const selectableTarget = findParentByFeature(target, isSelectable);
         if (selectableTarget) {
             isTargetSelected = selectableTarget.selected;
             selectableTarget.selected = true;
         }
         const restoreSelection = () => { if (selectableTarget) selectableTarget.selected = isTargetSelected; };
-        const menuItems = await this.menuProvider.getItems(target.root, mousePosition);
-        menuService.show(menuItems, mousePosition, restoreSelection);
+
+        if(isSelectable(target)){
+            if(isSelected(target)){
+                const menuItems = await this.menuProvider.getItems(target.root, mousePosition);
+                menuService.show(menuItems, mousePosition, restoreSelection);
+            }else{
+                const options = {selectedElementsIDs: [id], deselectedElementsIDs: Array.from(root.index.all().filter(isSelected), (val) => {return val.id;})};
+                this.actionDispatcher.dispatch(SelectAction.create(options)).then(() => {
```

Comment:
The method is `async`, so you can use `await` here instead of a callback chain.

Reply:
Hey thanks for the reply. Should all be fine now. 
Best Regards